contracts/credits/src/msg.rs

```diff
@@ -0,0 +1,154 @@
+use crate::state::Allocation;
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{Addr, Timestamp, Uint128};
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    /// DAO contract address
+    pub dao_address: String,
+    /// Airdrop contract address
+    pub airdrop_address: Option<String>,
+    /// Lockdrop contract address,
+    pub lockdrop_address: Option<String>,
+    /// When can start withdrawing NTRN funds
+    pub when_withdrawable: Timestamp,
+}
+
+#[cw_serde]
+pub enum ExecuteMsg {
+    /// UpdateConfig is a message to initialize all addresses.
+    /// Needed because there are circle deps between contracts.
+    /// [Permissioned - DAO]
+    UpdateConfig {
+        airdrop_address: String,
+        lockdrop_address: String,
+    },
+    /// AddVesting is a message that allows address to claim particular amount of NTRNs at particular time.
+    /// Can only store one vesting amount per address.
+    /// [Permissioned - Airdrop address]
+    AddVesting {
+        address: String,
+        amount: Uint128,
+        start_time: u64,
+        duration: u64,
+    },
+    /// Transfer is a base message to move tokens to another account without triggering actions.
+    /// [Permissioned - Airdrop address]
+    Transfer { recipient: String, amount: Uint128 },
+    /// Withdraw is a message that burns all vested CNTRN tokens
+    /// on the sender and sends NTRN tokens in 1:1 proportion.
+    /// [Permissionless]
+    Withdraw {},
+    /// Burn is a message only for airdrop account to destroy
```

Comment:
destroy certain amount of CNTRN's forever? I believe `burn` is a good enough known term to be used freely here. wyt?

Reply:
I added more description everywhere since it does more than original cw20, - namely we send untrns to accounts in return

Reply:
no no, the additional description is cool, I'm only not sure about the `destroy certain amount of CNTRN's forever` — this is what is usually understood as `burn`. I mean you can simply say `Burns certain amount of CNTRN's forever and send NTRN tokens to the caller in 1:1 proportion`